core/src/Processors/Resource/GetToolbar.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/*
+ * This file is part of MODX Revolution.
+ *
+ * Copyright (c) MODX, LLC. All Rights Reserved.
+ *
+ * For complete copyright and license information, see the COPYRIGHT and LICENSE
+ * files found in the top-level directory of this distribution.
+ */
+
+namespace MODX\Revolution\Processors\Resource;
+
+use MODX\Revolution\modProcessor;
+use MODX\Revolution\modResource;
+
+/**
+ * Gets a dynamic toolbar for the Resource tree.
+ */
+class GetToolbar extends modProcessor
+{
+    public function checkPermissions()
+    {
+        return $this->modx->hasPermission('resource_tree');
+    }
+
+    public function getLanguageTopics()
+    {
+        return ['resource', 'trash'];
+    }
+
+    public function process()
+    {
+        $items = [];
+        $context = '&context_key=' . $this->modx->getOption('default_context');
+        if ($this->modx->hasPermission('new_document')) {
+            $items[] = [
+                'cls' => 'tree-new-resource',
+                'tooltip' => $this->modx->lexicon('document_new'),
+                'handler' => 'new Function("this.redirect(\"?a=resource/create' . $context . '\");");',
+            ];
+        }
+        if ($this->modx->hasPermission('new_weblink')) {
+            $items[] = [
+                'cls' => 'tree-new-weblink',
+                'tooltip' => $this->modx->lexicon('add_weblink'),
+                'handler' => 'new Function("this.redirect(\"?a=resource/create&class_key=modWebLink' . $context . '\");");',
```

Comment:
How do we deal with `class_key` values like here and several other places in this class?